docs/brayanPolo/oop.md

```diff
@@ -0,0 +1,225 @@
+
+Encapsulation allows us to control who can view and use the various internal modules of our system. In terms of classes, encapsulation defines access to the members of the class.
+
+In C#, we can use access modifiers to specify control over external agents on different parts of our system, such as classes, class members, interfaces, among others. Suppose we have a variable called "speed," which we want to include in our "Car" class to indicate the speed at which a particular vehicle is moving. However, we only want the value of this variable to be visible and modifiable within the class. This can be achieved using a field or a property. Let's choose to do it with a property:
+
+
+>public class Car
+{
+public string Brand;
+public int YearReleasedToMarket { get; set; }
+private int Speed { get; set; }
+
+    public void Accelerate()
+    {
+        Speed += 10;
+    }
+>}
+
+When we instantiate the "Car" class, we cannot access the value of the "Speed" property, nor can we alter it from outside. What we can do is use the "Accelerate" function to increase the speed value by 10 units. This is one of the advantages of encapsulation: it allows us to control how the internal data of our object will be altered.
+
+If we want external agents to see the value of the "Speed" property but not be able to freely alter that value, we can use the following syntax:
+
+>public int Speed { get; private set; }
+
+## Inheritance
+
+Code sharing is a crucial feature in any software project, as it allows saving work when making changes to the system and enables a single algorithm to process different classes of entities, among other advantages.
+
+There are various ways to share code, and one of them is through the use of inheritance. Inheritance establishes a special relationship between two classes: the base class and the derived class. In this relationship, the derived class gains the ability to use specific properties and functionalities of the base class, even having the possibility to override the functionality of the latter. The main idea is that the derived class "inherits" some of the characteristics of the base class.
+
+We can illustrate this with an example using the "Car" class. A car is considered a type of vehicle, and besides, we want to process other types of vehicles, such as a truck. Both the car and the truck share the concept of speed, and both have the ability to accelerate and reverse. However, when a truck reverses, it must emit a sound. Finally, a car must have the ability to turn on the radio. Let's proceed to model this:
+
+>public class Vehicle
+{
+public string Brand;
```

Comment:
This is C# code. The assignment specification is about to provide Java code examples.